components/formats-gpl/src/loci/formats/in/TCSReader.java

```diff
@@ -247,6 +247,7 @@ protected void initFile(String id) throws FormatException, IOException {
     Arrays.sort(list);
 
     boolean isXML = checkSuffix(id, XML_SUFFIX);
+    if (isXML) xmlFile = l.getAbsolutePath();
```

Comment:
Is this effectively redundant with the call after `initFile(new Location(parent, file).getAbsolutePath())` below?